### DISCUSSION TITLE
How not to fetch attachments with body

### DISCUSSION THREAD

**Discussion by shoaib30** [mjs/imapclient#431] [Q&A] [answered]
Hi All,

I am new to IMAP and this may not be a imapclient specific question. I was wondering if there is a way to skip fetching attachments?

I know passing 'ENVELOPE' (`client.fetch(uid, ['ENVELOPE'])`) gives a summary, what I need is along with ENVELOPE is the text/plain and text/html (if present)

I am trying to fetch my entire mailbox to perform some analytics, it runs into a couple of gigabytes if I fetch along with attachments, and they are of no use to me.

any help would be appreciated.

**Comment by mjs** [comment, marked as answer]
This is completely possible by fetching the `BODYSTRUCTURE` for each message. This provides details for each part of the email, including the content type for each. Using this you can figure out the parts you're interested in (i.e. the text/plain or text/html parts), compute the part number(s) of interest and then fetch `BODY.PEEK[<part number>]` to retrieve it.

Relevant parts of IMAP RFC:
- https://tools.ietf.org/html/rfc3501#page-57
- https://tools.ietf.org/html/rfc3501#page-74
- https://tools.ietf.org/html/rfc3501#page-54

**Reply by shoaib30** [reply]
this helps a lot, thank you so much
